week9/answers.sql

```diff
@@ -1,3 +1,86 @@
 -- Create your tables, views, functions and procedures here!
 CREATE SCHEMA social;
 USE social;
+
+-- the tables:
+CREATE TABLE users (
+    user_id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    first_name VARCHAR(50),
+    last_name VARCHAR(50),
+    email VARCHAR(100),
+    created_on TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE TABLE sessions (
+    session_id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    user_id INT UNSIGNED,
+    created_on TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_on TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+    FOREIGN KEY (user_id) REFERENCES users(user_id)
+);
+
+CREATE TABLE friends (
+    user_friend_id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    user_id INT UNSIGNED,
+    friend_id INT UNSIGNED,
+    FOREIGN KEY (user_id) REFERENCES users(user_id),
+    FOREIGN KEY (friend_id) REFERENCES users(user_id)
+);
+
+CREATE TABLE posts (
+    post_id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    user_id INT UNSIGNED,
+    created_on TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_on TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+    content TEXT,
+    FOREIGN KEY (user_id) REFERENCES users(user_id)
+);
+
+CREATE TABLE notifications (
+    notification_id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    user_id INT UNSIGNED,
+    post_id INT UNSIGNED,
+    FOREIGN KEY (user_id) REFERENCES users(user_id),
+    FOREIGN KEY (post_id) REFERENCES posts(post_id)
+);
+
+-- the view: - Not sure what to order these by, so I'l wait and see if I get FAILS.
+CREATE VIEW notification_posts AS
+	SELECT n.user_id, u.first_name, u.last_name, p.post_id, p.content 
+    FROM notifications AS n
+	LEFT JOIN users AS u ON n.user_id = u.user_id
+	LEFT JOIN posts AS p ON n.post_id = p.post_id;
+
+-- When a new user is added, create a notification for everyone that states "{first_name} 
+-- {last_name} just joined!" (for example: "Jeromy Streets just joined!").
+DELIMITER ;;
+CREATE PROCEDURE add_new_user(IN first_name VARCHAR(30), IN last_name VARCHAR(30), IN email VARCHAR(30))
```

Comment:
This should be a TRIGGER so that when a user is added, all of the functionality will automatically occur.